README.md

```diff
@@ -73,314 +149,510 @@ public class MyApplication extends Application {
+            JsonObject dataObj = new JsonParser().parse(data).getAsJsonObject();
+            String type = dataObj.get("type").getAsString();
+            
+            return !isSystemMessage
+                && !EVENT_TYPE_ASSIGN.equals(type)
+                && !EVENT_TYPE_TRANSFER.equals(type)
+                && !EVENT_TYPE_CLOSE.equals(type);
+        }
+    }
+    
+    return true;
+}
+```
 
-ticket.setCustomFields(customFields, new Ticket.SetCustomFieldHandler() {
+### Request confirmation to close a ticket
+
+While admins have permission to directly close a ticket, agents can either close a ticket as admins do or ask customers whether to close a ticket, depending on the agent permission setting. The confirmation request message can have 3 types of state as below.
+
+|State|Description|
+|---|---|
+|WAITING |Set when an agent sends a confirmation request message. |
+|CONFIRMED| Set when a customer agrees to close the ticket. (Default: **true**)|
+|DECLINED |Set when a customer declines to close the ticket. (Default: **false**)|
+
+When a customer replies to the message, the response true (agree) or false (decline) is sent to Sendbird server as `CONFIRMED` or `DECLINED` by calling the `Ticket.confirmEndOfChat()` method.
+
+```java
+ticket.confirmEndOfChat(USER_MESSAGE, true|false, new Ticket.ConfirmEndOfChatHandler() {
     @Override
     public void onResult(Ticket ticket, SendBirdException e) {
-        if (e != null) {
-            // Error handling.
+        if (e != null) {    // Error.
             return;
         }
         
-        // ticket's customFields is rightly set
-        // (or a certain key could get ignored if the key is not defined yet)
+        // You can update the UI of the message. For example, you can hide YES and No buttons.
     }
 });
 ```
 
-## Setting Ticket priority
+```json
+{
+    "type": "SENDBIRD_DESK_INQUIRE_TICKET_CLOSURE",
+    "body": {
+        "state": "CONFIRMED"
+    }
+}
+```
+
+### Request customer feedback
+
+You can send a message to customers right after closing a ticket to ask whether they are satisfied with the support provided through the ticket. When the **Customer satisfaction rating** feature is turned on in your dashboard, customers will get a message asking to give a score and leave a comment as feedback. The message can have 2 states as below.
 
-Ticket information could be kept in `Priority`. 
-`setTicketPriority()` in `Ticket` lets the SDK set the `Priority` of the current ticket. 
-If you didn't set the Priority of ticket the default `Priority` has been set to `Priority.MEDIUM`
+|State|Description|
+|---|---|
+|WAITING|Set when an agent sends a customer feedback request message.|
+|CONFIRMED|Set when a customer sends a response.|
 
+When a customer replies to the message, their score and comment for the ticket are sent to the Desk server by calling the `Ticket.submitFeedback()` method. Then, the state of the confirmation request message is changed to `CONFIRMED`.
```

Comment:
`Ticket.submitFeedback()` -> `ticket.submitFeedback()` (lowercased t as `submitFeedback()` is not a static method)

Reply:
Done!